### ISSUE TITLE
Remove IStreamClientEndpointStringParser

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8094]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8094|
|Type|task|
|Created|2015-11-07 11:15:15Z|
|Branch|https://github.com/twisted/twisted/tree/IStreamClientEndpointStringParser-rip-8094|

It's been deprecated for 5 years, time to remove it.

<details><summary>Searchable metadata</summary>

```
trac-id__8094 8094
type__task task
reporter__hawkowl hawkowl
priority__normal normal
milestone__None None
branch__branches_IStreamClientEndpointStringParser_rip_8094 branches/IStreamClientEndpointStringParser-rip-8094
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1446894915681755 1446894915681755
changetime__1447413413840622 1447413413840622
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [46100]) Branching to IStreamClientEndpointStringParser-rip-8094.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Builds are looking reasonable. Please review.

Part of [#8092](https://github.com/twisted/twisted/issues/8092)

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @hawkowl|
|-|-|

Looks good. Thanks!

Please merge!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl set status to `closed`|
|-|-|

(In [46177]) Merge IStreamClientEndpointStringParser-rip-8094: Remove IStreamClientEndpointStringParser.

Author: hawkowl
Reviewer: adiroiban
Fixes: [#8094](https://github.com/twisted/twisted/issues/8094)
